fl4health/datasets/skin_cancer/preprocess_skin.py

```diff
@@ -0,0 +1,281 @@
+    vienna_data = HAM_df[HAM_df["dataset"] != "rosendahl"]
+    vienna_data.to_csv(os.path.join(HAM_10000_path, "HAM_vienna.csv"), mode="w")
+
+    # Load PAD-UFES-20
+    PAD_UFES_20_path = f"{data_path}/PAD-UFES-20"
+    PAD_UFES_20_csv_path = os.path.join(PAD_UFES_20_path, "metadata.csv")
+    PAD_UFES_20 = pd.read_csv(PAD_UFES_20_csv_path)
+
+    # Load Derm7pt
+    Derm7pt_path = f"{data_path}/Derm7pt"
+    Derm7pt_csv_path = os.path.join(Derm7pt_path, "meta/meta.csv")
+    Derm7pt = pd.read_csv(Derm7pt_csv_path)
+
+    derm7pt_labelmap = {
+        "lentigo": "MISC",
+        "melanosis": "MISC",
+        "miscellaneous": "MISC",
+    }
+
+    MISC_list = ["lentigo", "melanosis", "miscellaneous"]
+    Derm7pt_core = Derm7pt[~Derm7pt["diagnosis"].isin(MISC_list)]
+    Derm7pt_core.to_csv(os.path.join(Derm7pt_path, "meta/meta_core.csv"))
+
+    ###########################################################################
+    # Unify the naming of labels in the below section
+
+    official_columns = ["MEL", "NV", "BCC", "AK", "BKL", "DF", "VASC", "SCC"]
+
+    # Client: Barcelona
+    ###########################################################################
+    ISIC_2019_data_path = f"{data_path}/ISIC_2019/ISIC_2019_Training_Input"
+    ISIC_csv_path = os.path.join(ISIC_2019_path, "ISIC_2019_core.csv")
+    Barcelona_df = pd.read_csv(ISIC_csv_path)
+    Barcelona_new = Barcelona_df[["image", "MEL", "NV", "BCC", "AK", "BKL", "DF", "VASC", "SCC", "UNK"]]
+    preprocessed_data: Dict[str, Any] = {
+        "columns": official_columns,
+        "original_columns": official_columns,
+        "data": [],
+    }
+
+    for i in range(len(Barcelona_new)):
+        temp = list(Barcelona_new.loc[i].values[:-1])
+        img_path = os.path.join(ISIC_2019_data_path, temp[0] + ".jpg")
+        origin_labels = temp[1:]
+        extended_labels = temp[1:]
+        preprocessed_data["data"].append(
+            {
+                "img_path": img_path,
+                "origin_labels": origin_labels,
+                "extended_labels": extended_labels,
+            }
+        )
+
+    file_path = os.path.join(f"{data_path}/ISIC_2019", "ISIC_19_Barcelona.json")
+    save_to_json(preprocessed_data, file_path)
+
+    # Client: Rosendahl
+    ###########################################################################
+    HAM_columns = ["MEL", "NV", "BCC", "AK", "BKL", "DF", "VASC"]
+
+    HAM_10000_data_path = f"{data_path}/HAM10000"
+    rosendahl_df = pd.read_csv(os.path.join(HAM_10000_path, "HAM_rosendahl.csv"))
+    vienna_df = pd.read_csv(os.path.join(HAM_10000_path, "HAM_vienna.csv"))
+
+    ham_labelmap = {
+        "akiec": "AK",
+        "bcc": "BCC",
+        "bkl": "BKL",
+        "df": "DF",
+        "mel": "MEL",
+        "nv": "NV",
+        "vasc": "VASC",
+    }
+
+    preprocessed_data = {
+        "columns": official_columns,
+        "original_columns": HAM_columns,
```

Comment:
These columns aren't the "original columns but rather the new ones? Should we be storing the originals here?

Reply:
i think for the code, original columns means the original dataset columns those dataset had